File: gerenciador de tarefas/models/Lista.php
<?php 

class Lista{
    private $titulo;
    private $itens;
	private $ID;
	private static $contadorIds = 0;

    public function __construct($titulo){
		self::$contadorIds++;
        $this->titulo = $titulo;
        $this->itens = array();
    }

	public function getTitulo() {
		return $this->titulo;
	}
	
	public function setTitulo($titulo): self {
		$this->titulo = $titulo;
		return $this;
	}

	public function getItens() {
		return $this->itens;
	}
	
	public function setItens($itens): self {
		$this->itens = $itens;
		return $this;
	}

    public function adicionarItem($item){
		$this->itens[] = $item;
    }
    
    public function excluirItem($indice)
    {
        if (isset($this->itens[$indice])) {
            unset($this->itens[$indice]);
            // Importante: após remover o item da lista, você pode querer reindexar o array para evitar índices não sequenciais.
            // Caso não se preocupe com os índices, isso não é obrigatório.
            $this->itens = array_values($this->itens);
        }
    }

	public function getID() {
		return $this->ID;
	}
	
	public function setID($ID): self {
		$this->ID = $ID;
		return $this;
	}
}
?>

File: gerenciador-de-tarefas-spring/gerenciador-de-tarefas/src/main/java/com/tanyrm/gerenciadordetarefas/model/Lista.java
package model;

import java.util.ArrayList;

public class Lista {
    private String titulo;
    private ArrayList<Item> itens;

    public Lista (String titulo){
        this.titulo = titulo;
        this.itens = new ArrayList<Item>();
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public ArrayList<Item> getItens() {
        return itens;
    }

    public void adicionarItem(Item item){
        this.itens.add(item);
    }

    public void excluirItem(Item item){
        this.itens.remove(item);
    }
}
